proc-datacite continually throwing exceptions after minting a test DOI

After minting a DOI as anonymous user and filling all fields, `var/log/ezid/background_jobs/ezid-proc-error.log` reports:

```
an 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: DEBUG ezidapp.management.commands.proc_base proc_base 26975 140261853968192 DataCite: Processing task: create
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: ERROR ezidapp.management.commands.proc_base proc_base 26975 140261853968192 DataCite: Task registration error
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: Traceback (most recent call last):
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: File "/apps/ezid/ezid/ezidapp/management/commands/proc_base.py", line 94, in run
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: self.do_task(task_model)
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: File "/apps/ezid/ezid/ezidapp/management/commands/proc_base.py", line 166, in do_task
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: self.create(task_model)
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: File "/apps/ezid/ezid/ezidapp/management/commands/proc-datacite.py", line 25, in create
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: self._overwrite(task_model)
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: File "/apps/ezid/ezid/ezidapp/management/commands/proc-datacite.py", line 44, in _overwrite
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: assert type(r) is not str, f"Unexpected return: {repr(r)}"
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: AssertionError: Unexpected return: 'DOI metadata requirements not satisfied: no publisher'
Jan 27 05:24:56 uc3-ezidui02x2-stg ezid-proc-errorlog: DEBUG ezidapp.management.commands.proc_base proc_base 26975 140261853968192 DataCite: Closing DB connections and sleeping for 1.00s...
```

--- a/ezidapp/management/commands/proc-datacite.py
+++ b/ezidapp/management/commands/proc-datacite.py
@@ -41,8 +41,22 @@ class Command(ezidapp.management.commands.proc_base.AsyncProcessingCommand):
         metadata = ref_id.metadata
         datacenter = ref_id.datacenter
         r = impl.datacite.uploadMetadata(doi, {}, metadata, True, datacenter)
-        assert type(r) is not str, f"Unexpected return: {repr(r)}"
+        # r can be:
+        # None == success
+        # string == error message, something wrong with payload
+        # exception == something else went wrong.
+        # Exceptions are handled in the outer run method, so don't handle here.
+        #
+        if r is not None:
+            log.error("datacite.uploadMetadata returned: %s", r)
+            raise ezidapp.management.commands.proc_base.AsyncProcessingError(r)
         r = impl.datacite.setTargetUrl(doi, ref_id.target, datacenter)
-        assert type(r) is not str, f"Unexpected return: {repr(r)}"
+        # r can be:
+        #   None on success
+        #   a string error message if the target URL was not accepted by DataCite
+        #   a thrown exception on other error.
+        if r is not None:
+            log.error("datacite.setTargetUrl returned: %s", r)
+            raise ezidapp.management.commands.proc_base.AsyncProcessingError(r)
         if metadata.get("_is", "public") != "public" or metadata.get("_x", "yes") != "yes":
             impl.datacite.deactivateIdentifier(doi, datacenter)

--- a/ezidapp/management/commands/proc_base.py
+++ b/ezidapp/management/commands/proc_base.py
@@ -21,6 +21,10 @@ import ezidapp.models.async_queue
 
 log = logging.getLogger(__name__)
 
+class AsyncProcessingError(Exception):
+    """Raise when a permanent error is encountered.
+    """
+    pass
 
 class AsyncProcessingCommand(django.core.management.BaseCommand):
     help = __doc__
@@ -178,6 +182,9 @@ class AsyncProcessingCommand(django.core.management.BaseCommand):
 
     def is_permanent_error(self, e):
         """Return True if exception appears to be due to a permanent error"""
+        # The error is raised from something in a processor
+        if isinstance(e, AsyncProcessingError):
+            return True
         # We were able to connect to the server, but it returned a 5xx response
         if isinstance(e, urllib.error.HTTPError) and e.code >= 500:
             return True

